src/shared/components/workbasket-filters/workbasket-filters.component.ts

```diff
@@ -0,0 +1,239 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ActivatedRoute, ActivatedRouteSnapshot } from '@angular/router';
+import { FormGroup } from '@angular/forms';
+import 'rxjs/add/operator/do';
+import { Jurisdiction, CaseState, CaseTypeLite, WorkbasketInputModel } from '../../domain';
+import { READ_ACCESS } from '../../domain/case-view/access-types.model';
+import { JurisdictionService, AlertService, WindowService, OrderService, WorkbasketInputFilterService,
+  DefinitionsService } from '../../services';
+
+const FORM_GROUP_VAL_LOC_STORAGE = 'workbasket-filter-form-group-value';
+const SAVED_QUERY_PARAM_LOC_STORAGE = 'savedQueryParams';
+@Component({
+  selector: 'ccd-workbasket-filters',
+  templateUrl: './workbasket-filters.component.html',
+  styleUrls: ['./workbasket-filters.component.scss']
+})
+export class WorkbasketFiltersComponent implements OnInit {
+
+  public static readonly PARAM_JURISDICTION = 'jurisdiction';
+  public static readonly PARAM_CASE_TYPE = 'case-type';
+  public static readonly PARAM_CASE_STATE = 'case-state';
+
+  @Input()
+  defaults;
+
+  @Output()
+  onApply: EventEmitter<any> = new EventEmitter();
+
+  @Output()
+  onReset: EventEmitter<any> = new EventEmitter();
+
+  workbasketInputs: WorkbasketInputModel[];
+  workbasketInputsReady: boolean;
+
+  selected: {
+    init?: boolean,
+    jurisdiction?: Jurisdiction,
+    caseType?: CaseTypeLite,
+    caseState?: CaseState,
+    formGroup?: FormGroup,
+    page?: number,
+    metadataFields?: string[]
+  };
+
+  formGroup: FormGroup = new FormGroup({});
+
+  jurisdictions: Jurisdiction[];
+  selectedJurisdictionCaseTypes?: CaseTypeLite[];
+  selectedCaseTypeStates?: CaseState[];
+
+  initialised = false;
+
+  constructor(
+    private route: ActivatedRoute,
+    private workbasketInputFilterService: WorkbasketInputFilterService,
+    private orderService: OrderService,
+    private jurisdictionService: JurisdictionService,
+    private alertService: AlertService,
+    private windowService: WindowService,
+    private definitionsService: DefinitionsService,
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.selected = {};
+
+    this.definitionsService.getJurisdictions(READ_ACCESS)
```

Comment:
Remember man web has to use the old component due to performance reasons. Are we extracting the old component as well having two effectively or we moved this one and changed?

Reply:
Are we going to have two components then? Or will the component decide whether to use the service?

Reply:
I think two components is better as we do not need to regression test man web that way so will save us time. It would make the extraction consistent with all other ones this way. It would just be a new wrapper component then.

Reply:
Cool, I'll just rename this component then.

Reply:
Done